src/Phug/Reader.php

```diff
@@ -408,7 +412,8 @@ public function consume($length = null)
         }
 
         $consumedPart = mb_substr($this->input, 0, $length, $this->encoding);
-        $this->input = mb_substr($this->input, $length, mb_strlen($this->input) - $length, $this->encoding);
+        $this->input = substr($this->input, strlen($consumedPart));
```

Comment:
I went too fast, this part cannot be replaced this way, it will cut mb characters. We must keep mb_substr/mb_strlen to fully support UTF-8 strings even if it cost some perf. But caching the inputLength is a very good optimization yet.